api/v1/submissions/PKPSubmissionController.php

```diff
@@ -819,6 +827,50 @@ public function delete(Request $illuminateRequest): JsonResponse
         return response()->json($submissionProps, Response::HTTP_OK);
     }
 
+    /**
+     * Change submission language
+     */
+    public function changeLocale(Request $illuminateRequest): JsonResponse
+    {
+        $publication = Repo::publication()->get((int) $illuminateRequest->route('publicationId'));
+
+        if (!$publication) {
+            return response()->json([
+                'error' => __('api.404.resourceNotFound'),
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return response()->json([
+                'error' => __('api.publications.403.submissionsDidNotMatch'),
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        $paramsSubmission = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_SUBMISSION, $illuminateRequest->input());
+        $newLocale = $paramsSubmission['locale'] ?? null;
+
+        // Submission language can not be changed when there are more than one publication or a publication's status is published
+        if (!$newLocale || count($submission->getData('publications')) > 1 || $publication->getData('status') === PKPSubmission::STATUS_PUBLISHED) {
+            return response()->json(['error' => __('api.submission.403.cantChangeSubmissionLanguage')], Response::HTTP_FORBIDDEN);
+        }
+
+        // Convert a form field value to multilingual (if it is not) and merge rest values
+        collect(app()->get('schema')->getMultilingualProps(PKPSchemaService::SCHEMA_PUBLICATION))
+            ->each(fn (string $p) => $illuminateRequest->whenHas($p, fn ($v) => $illuminateRequest->merge([$p => array_merge($publication->getData($p) ?? [], is_array($v) ? $v : [$newLocale => $v])])));
```

Comment:
With this using deeply nested arrow functions, the single-letter variables and everything being on one line make it kind of difficult to parse what's going on. I would recommend more explicit variable names and possibly breaking the `each` into more than one line for readability.